### ISSUE TITLE
Allow override of tag name for display

### ISSUE THREAD

**Issue by kcrisman** [rbeezer/mathbook#39]
As requested in #21, I am asking for a `<comment>` tag.  In this case, it should not be something numbered - that would be a `<remark>`.  Ideally it could be marked up in some way with a custom style sheet.   Comments are just asides that need to be clearly marked as asides; that is not really possible in the usual `<p>` format, as parentheses around a paragraph is not a strong indication.

(Alternately, one could have a switch on various things allowing them to not be numbered; an unnumbered remark is a comment, in my current thinking, though probably not universally.)

**Comment by rbeezer** [comment]
And when rendered, could you imagine it being a "marginal note" via LaTeX and 
off in the right margin whitespace via HTML?

In what ways do you intend this to be different than a footnote (which is 
implemented)?

(Was that a meta-example at the end?)

Thanks,
Rob

**Comment by kcrisman** [comment]
No meta-examples.  Basically, I just want do something David would say is bad - to be able to have

Main content main content Main content main content Main content main content Main content main content Main content main content Main content main content 

**Comment** you can fry fish for supper tonight

Main content main content Main content main content Main content main content Main content main content 

Note the bold.  But as I say, having a remark one could leave unnumbered (while others might perhaps be unnumbered) and perhaps renamed would be fine too.  I don't know about marginal notes, I've never tried those.  However, maybe that's what this could do, certainly I've seen them in calc books (e.g. 'historical comment').

What I personally want is just to be able to do bold something-or-other that is highlighted but then also styleable as "not the normal flow, but too important to be a footnote".  If I can get that with an unnumbered remark I can rename as comment, great.  Your idea about a comment in the margins sounds good too, it's just not the thing I was thinking of.

Basically, allowing some items to be dropped from the numbering hierarchy at will (perhaps `<tag number='false'>`) and renamed (`<tag number='false' name='my friend Sally'>`) would help tons, but that may not be the vision you have.

**Comment by davidfarmer** [comment]
I have found that some people use the comment environment in LaTeX in the sense of a
comment in computer code: the comment is visible in the source bit has no effect on the output.

This surprised me, and led to some weird output of SL2X.  But we need to be aware that
"comment" means different things to different people.  So I suggest we should not use < comment >
to tag something that should appear as a Comment in the output.

Since "remark" is pretty similar to "comment" in the context of this thread, I have one suggestion
and one question.

My suggestion, which I think is the same as what is suggested at the end of the previous
comment, is that everything that is numbered by default should have an un-numbered version,
and using that feature should be as easy as adding the "*" in LaTeX.  Not sure that
"number = 'false'" is the best way to do it, but it should be easy

My question, which also may be a point form the previous comment, is to what extent we should
allow the author to over-ride the "name" of an environment?  Let's say there is no environment
named "Conundrum", but you want that environment and you think it should be typeset just
like a Conjecture, except for using the word Conundrum instead.  Should you be able to
use the < conjecture> environment, and just over-ride the name?

**Comment by kcrisman** [comment]
This is basically what I was suggesting, only much better formulated.  I'll open a new thing for the unnumbered, and leave this for the overriding.

**Comment by rbeezer** [comment]
This became the "override-the-name" issue.  You can now do that globally.  See 31a2bddc and upcoming support post.  In particular, yu could turn every "Remark" into a "Comment" (though then you would not be abl to have any remarks anymore).  Optional no-number is part of a larger project latr.

**Comment by kcrisman** [comment]
Nice!

But what about a once-off renaming?  Or is that just too far outside the design philosophy?  E.g. suppose one uses all possible tags, and still wants

**Proposition 5.2** Stuff

**Proposition 5.3** Stuff

**Wacky Out of Control Fact (5.4?)** lots of detail

**Proposition 5.4** (or possibly 5.5)

**Comment by rbeezer** [comment]
Definitely way, way, way too far outside design philosophy.

Any semblance of meaning or consistency is lost, so you might as well just use 
LaTeX and whatever HTML-to-LaTeX translator you can bend to your will.

I will be adding "problem" and "question" so I expect there will almost-always 
be enough unused environments around to recycle for a different purpose.

What I already implemented is outside design philosophy.  So once-off renaming 
is never going to happen.  ;-)

Rob

**Comment by kcrisman** [comment]
Or, I suppose, one could hand-edit the source files.  Too bad, as I really enjoy reading papers and books that have things like "Main Lemma".  It makes it easier to follow when something is unusual.

And how does FCLA fit into this, with its very nonstandard "numbering", anyway?

**Comment by rbeezer** [comment]
Do you think the following would be abused?

`<lemma number="no"><title>Main Lemma</title>...</lemma>`

and the no-number/has-title combination gets recognized so at an `xref` to this lemma will be the title, while the lemma itself will just have the title (not Lemma (Main Lemma)).

FCLA will be an example of minimal work for massive customization.  Every object will have an `acro="XYZ"` attribute and one or two templates will be overridden in a customization layer with very simple replacement templates reporting the acronym as the "number".  You could then ditch the customization and get numbers, but I do not plan to distribute such a version.  I *like* my acronyms.

**Comment by kcrisman** [comment]
I am sure it will be abused, but so will any other customizable feature.  Abuse of this is for editors to say "whoa there".  It sounds, then, like FCLA is a perfect example of that!  But my point is that at least for certain things it should be made easier than having the customization layer, and if you have various dire warnings of eternal doom in the example demonstrating this and pointing to guidelines for authors, we should survive.
